homeassistant/components/tradfri.py

```diff
@@ -58,13 +57,27 @@ def configuration_callback(callback_data):
         """Handle the submitted configuration."""
         try:
             from pytradfri.api.aiocoap_api import APIFactory
+            from pytradfri import RequestError
+            from uuid import uuid4
         except ImportError:
             _LOGGER.exception("Looks like something isn't installed!")
             return
 
-        api_factory = APIFactory(host, psk_id=GATEWAY_IDENTITY)
-        psk = yield from api_factory.generate_psk(callback_data.get('key'))
-        res = yield from _setup_gateway(hass, config, host, psk,
+        identity = uuid4().hex
+        security_code = callback_data.get('security_code')
+
+        api_factory = APIFactory(host, psk_id=identity, loop=hass.loop)
+        # Need To Fix: currently entering a wrong security code sends
+        # pytradfri aiocoap API into an endless loop.
+        # Should just raise a requestError or something.
+        try:
+            key = yield from api_factory.generate_psk(security_code)
+        except RequestError:
+            hass.async_add_job(configurator.notify_errors, instance,
```

Comment:
Just call `async_notify_errors` instead of async_add_job…

Reply:
Will change. Was an existing call by the way.